### PR TITLE
Proposal: driver spec v1

### PR DESCRIPTION
**Pull request by ehazlett**
This is a draft of a v1 driver spec.  This describes what a driver needs to provide to be supported by Machine as well as the responsibilities by the contributor.

### CODE REVIEW

**Comment by ehazlett** [comment]
This is to get the ball rolling.  From this we can get the technical bits (proper interfaces, etc).  PTAL @bfirsh @sthulb @nathanleclaire @cjellick @frapposelli @ggiamarchi

**Comment by sthulb** [comment]
We could perhaps explicitly state that the driver is only responsible for hypervisor operations. No provisioning should be done by the driver.

**Comment by nathanleclaire** [comment]
I think I would prefer it to be in `docs` folder so it is easier to discover (don't really care for docs scattered throughout the repo).

In terms of content: LGTM.  I would prefer to get this in and then iterate on more detailed stuff in later PRs.

**Comment by sthulb** [comment]
@nathanleclaire +1

**Comment by ehazlett** [comment]
@nathanleclaire cool -- i'll move it to docs

**Comment by nathanleclaire** [comment]
LGTM

File: docs/DRIVER_SPEC.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+current state of the instance (running, stopped, error, etc).  This should
+return an error on failure.
+
+# Testing
+Testing is strongly recommended for drivers.  Unit tests are preferred as well
+as inclusion into the [integration tests](https://github.com/docker/machine#integration-tests).
+
+# Maintaining
+Driver contributors are strongly encouraged to maintain the driver to keep
+it supported.  We recommend and encourage contributors to join in the weekly
```
Comment (cjellick) [inline docs/DRIVER_SPEC.md]: Is there a web page you could link to with the details of the weekly meeting? Guessing people would be curious if they saw this.

Comment (ehazlett) [inline docs/DRIVER_SPEC.md]: We haven't started them yet.  I want to start them as soon as next week.  They will be open to all and the notes will be published.  We are just trying to figure out the best way to be consistent across the projects.  Once we do, I will update it.  Thanks!

**Comment by sthulb** [comment]
@ehazlett Can we add consistent method signatures to this? 

The `d` vs `driver` thing gets to me when changing things, we should standardise on it early :)

**Comment by ehazlett** [comment]
@stefanfoulis +1!

**Comment by ehazlett** [comment]
Merging.  We can iterate as we go.  A design will follow shortly.

**Pull request merged**